packages/sdk/src/client.ts

```diff
@@ -2285,6 +2270,24 @@ export class Client
         }
     }
 
+    async makeMediaEventWithHashAndAddToMediaStream(
+        creationCookie: CreationCookie,
+        payload: PlainMessage<StreamEvent>['payload'],
+    ): Promise<{ creationCookie: CreationCookie }> {
+        const streamIdStr = streamIdAsString(creationCookie.streamId)
+        check(isDefined(streamIdStr) && streamIdStr !== '', 'streamId must be defined')
+        const event = await makeEvent(this.signerContext, payload, creationCookie.prevMiniblockHash)
+
+        const resp = await this.rpcClient.addMediaEvent({
+            event,
+            creationCookie,
+        })
+
+        check(isDefined(resp.creationCookie), 'creationCookie not found in response')
```

Comment:
the original makeEventWithHashAndAddToStream has a retry loop... we don't need to retry any more? I guess that makes sense... can you add a doc string with a comment explaining how this function is different than the original?